future-apps/payroll/contracts/Payroll.sol

```diff
@@ -293,21 +289,24 @@ contract Payroll is EtherTokenConstant, IForwarder, IsContract, AragonApp {
      * @param _distribution Array with each token's corresponding proportions (must be integers summing to 100)
      */
     function determineAllocation(address[] _tokens, uint256[] _distribution) external employeeMatches nonReentrant {
-        // Check arrays match
+        // Check arrays length match
+        require(_tokens.length <= MAX_ALLOWED_TOKENS, ERROR_MAX_ALLOWED_TOKENS);
         require(_tokens.length == _distribution.length, ERROR_TOKEN_ALLOCATION_MISMATCH);
 
         uint256 employeeId = employeeIds[msg.sender];
         Employee storage employee = employees[employeeId];
 
-        // Delete previous allocation
-        for (uint256 j = 0; j < allowedTokensArray.length; j++) {
-            delete employee.allocation[allowedTokensArray[j]];
+        // Delete previous token allocations
+        address[] previousAllowedTokenAddresses = employee.allowedTokenAddresses;
```

Comment:
Actually, on second thought, I'm not so sure. Perhaps moving it all to memory is cheaper, since the length checks will likely be another SLOAD?

I think there's another instance where we do actually use `address [] storage`, and it may be good to do a quick test to see which one is more performant.

Reply:
Yea, I added some tests and it's cheaper copying all to memory, I actually modify `_ensureEmployeeTokenAllocationsIsValid` to do the same